core/tests/karma.conf.js

```diff
@@ -112,10 +112,8 @@ module.exports = function(config) {
       'karma-coverage'
     ],
     ngHtml2JsPreprocessor: {
-      cacheIdFromPath: function(filepath) {
-        return filepath;
-      },
-      moduleName: 'directiveTemplates'
+      moduleName: 'directiveTemplates',
+      prependPrefix: '/'
```

Comment:
Could you explain the rationale behind this change?

Reply:
Before, we saved directive HTML to `$templateCache` without `/` in front of the path and loaded them with `/` in front of the path. That was the reason why Karma tests working with directive HTML didn't work. Now we add the `/` in front of the path. All other changes in frontend tests are because of this.